src/commands/appSettings/downloadAppSettings.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebSiteManagementModels } from "azure-arm-website";
+import { DotenvParseOutput } from "dotenv";
+import * as fse from 'fs-extra';
+import * as os from 'os';
+import * as vscode from 'vscode';
+import { AppSettingsTreeItem, SiteClient } from "vscode-azureappservice";
+import { envFileName } from "../../constants";
+import { ext } from "../../extensionVariables";
+import * as workspaceUtil from '../../utils/workspace';
+import { getLocalEnvironmentVariables } from './getLocalEnvironmentVariables';
+
+export async function downloadAppSettings(node?: AppSettingsTreeItem): Promise<void> {
+    if (!node) {
+        node = <AppSettingsTreeItem>await ext.tree.showTreeItemPicker(AppSettingsTreeItem.contextValue);
+    }
+
+    const client: SiteClient = node.root.client;
+
+    const message: string = 'Select the destination file for your downloaded settings.';
+    const envVarPath: string = await workspaceUtil.selectWorkspaceFile(ext.ui, message, () => envFileName);
+    const envVarUri: vscode.Uri = vscode.Uri.file(envVarPath);
+
+    await node.runWithTemporaryDescription('Downloading...', async () => {
+        ext.outputChannel.appendLine(`Downloading settings from "${client.fullName}"...`);
+        const localEnvVariables: DotenvParseOutput = await getLocalEnvironmentVariables(envVarPath, true /* allowOverwrite */);
+        const remoteEnvVariables: WebSiteManagementModels.StringDictionary = await client.listApplicationSettings();
+        if (remoteEnvVariables.properties) {
+            await node.confirmOverwriteSettings(remoteEnvVariables.properties, localEnvVariables, envFileName);
+        }
+
+        await fse.ensureFile(envVarPath);
+        await fse.writeFile(envVarPath, convertAppSettingsToEnvVariables(localEnvVariables, client.fullName));
+    });
+
+    const doc: vscode.TextDocument = await vscode.workspace.openTextDocument(envVarUri);
+    await vscode.window.showTextDocument(doc);
```

Comment:
I think we need to discuss this. For example, I like seeing this file right away, but based on your recent PR to obfuscate settings by default, we might not want to open a file that has the same information as app settings.

Maybe we can "focus" on the file in the file view without actually opening it? Not sure if that's possible. Also you could do a notification that says "Finished downloading" or whatever with a "View file" button. I think that more closely mirrors the "Click to view" thing

Reply:
![image](https://user-images.githubusercontent.com/5290572/50035577-cab8b200-ffb7-11e8-9106-fd11355eedca.png)